### PR TITLE
Implement login with api

### LINKED ISSUE #9: access to api data
access to api data

### PR DESCRIPTION
**Pull request by satoi-fish**
Implement login with api and init token decode provider

image

Close #9

### CODE REVIEW

File: web/src/api/http-client.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  internalservererror = 500
+}
+
+export async function base<T>(url: string, method: "get" | "post", data?: object) {
+  const settings = (window as any).appSettings as AppSettings;
+  return await fetch(`${settings.serverUrl}${url}`, {
+    method: method,
+    body: data ? JSON.stringify(data) : undefined,
+    headers: {
+      js_version: settings.jsVersion,
```
Comment (ppXD) [inline web/src/api/http-client.tsx:29]: 暂时不用传

File: web/src/api/http-client.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      source_system: settings.sourceSystem,
+      Authorization: (window as any).token,
```
Comment (ppXD) [inline web/src/api/http-client.tsx:31]: 暂时不用传

File: web/src/api/http-client.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export async function base<T>(url: string, method: "get" | "post", data?: object) {
+  const settings = (window as any).appSettings as AppSettings;
+  return await fetch(`${settings.serverUrl}${url}`, {
+    method: method,
+    body: data ? JSON.stringify(data) : undefined,
+    headers: {
+      js_version: settings.jsVersion,
+      source_system: settings.sourceSystem,
```
Comment (ppXD) [inline web/src/api/http-client.tsx:30]: 暂时不用传

File: web/src/api/http-client.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export interface IResponse<T> {
+  code: ResponseCode;
+  msg: string;
+  data: T;
+}
+
+export enum ResponseCode {
+  ok = 200,
```
Comment (ppXD) [inline web/src/api/http-client.tsx:18]: 大写开头

File: web/src/api/http-client.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+export interface IResponse<T> {
+  code: ResponseCode;
+  msg: string;
+  data: T;
+}
+
+export enum ResponseCode {
+  Ok = 200,
+  Unauthorized = 401,
+  Internalservererror = 500
```
Comment (ppXD) [inline web/src/api/http-client.tsx:20]: InternalServerError

File: web/src/dtos/login/index.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export interface ILogin {
```
Comment (ppXD) [inline web/src/dtos/login/index.ts:1]: ILoginRequest

**Review by 0824F — APPROVED** [review]
LGTM

**Pull request merged**